### PR TITLE
[in-proc] Update nuget packages

### PR DESCRIPTION
**Pull request by jviau**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Updates and organizes some nuget packages.

Output of `Verify_DepsJsonChanges`:

```
IMPORTANT: The dependencies in WebHost have changed and MUST be reviewed before proceeding. Please follow up with brettsam, fabiocav or mathewc for approval.

Previous file: C:\repos\func\host-2\out\bin\WebJobs.Script.Tests\debug_net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json
New file:      C:\repos\func\host-2\out\bin\WebJobs.Script.WebHost\debug_net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json

  Changed:
    - Azure.Security.KeyVault.Secrets.dll: 4.2.0.0/4.200.21.31503 -> 4.6.0.0/4.600.24.11403

  Removed:

  Added:
    - System.Formats.Asn1.dll: 6.0.0.0/6.0.3224.31407
```

### CODE REVIEW

File: test/CSharpPrecompiledTestProjects/Dependency55/Dependency55.csproj [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.IdentityModel.Protocols" Version="5.5.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
```
Comment (fabiocav) [inline test/CSharpPrecompiledTestProjects/Dependency55/Dependency55.csproj:9]: Required?

**Review by fabiocav — COMMENTED** [review]
Was `System.Formats.Asn1` flagged as an addition and not a change by the test?

File: test/CSharpPrecompiledTestProjects/Dependency55/Dependency55.csproj (hunk 1 above)
Comment (jviau) [reply test/CSharpPrecompiledTestProjects/Dependency55/Dependency55.csproj:9]: Yes, this is a change that has been in `dev` for a bit. Forgot to port it over to `in-proc`. It's just to have nuget restore stop warning on this dependency.

**Comment by jviau** [comment]
> Was `System.Formats.Asn1` flagged as an addition and not a change by the test?

Yeah, that is the output of the test exactly. If you look at the deps file, it is not an addition. Maybe the test is confused as it is promoted to a direct dependency now?

**Review by fabiocav — APPROVED** [review]

**Review by surgupta-msft — APPROVED** [review]

**Pull request merged**